kernel/kernel-api/src/main/java/io/delta/kernel/engine/CommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,152 @@
+            String logPath,
+            long currentVersion,
+            AbstractMetadata currentMetadata,
+            AbstractProtocol currentProtocol);
+
+    /**
+     * API to commit the given set of actions to the table represented by logPath at the
+     * given commitVersion.
+     *
+     * @param logPath        The path to the delta log of the table that should be committed to.
+     * @param tableConf      The table configuration that was returned by the commit coordinator
+     *                       client during registration.
+     * @param commitVersion  The version of the commit that is being committed.
+     * @param actions        The actions that need to be committed.
+     * @param updatedActions The commit info and any metadata or protocol changes that are made
+     *                       as part of this commit.
+     * @return CommitResponse which contains the file status of the committed commit file. If the
+     * commit is already backfilled, then the file status could be omitted from the response
+     * and the client could retrieve the information by itself.
+     * @throws CommitFailedException if the commit failed.
+     */
+    CommitResponse commit(
+            String logPath,
+            Map<String, String> tableConf,
+            long commitVersion,
+            CloseableIterator<Row> actions,
+            UpdatedActions updatedActions) throws CommitFailedException;
+
+    /**
+     * API to get the unbackfilled commits for the table represented by the given logPath.
+     * Commits older than startVersion or newer than endVersion (if given) are ignored. The
+     * returned commits are contiguous and in ascending version order.
+     *
+     * Note that the first version returned by this API may not be equal to startVersion. This
+     * happens when some versions starting from startVersion have already been backfilled and so
+     * the commit coordinator may have stopped tracking them.
+     *
+     * The returned latestTableVersion is the maximum commit version ratified by the commit
+     * coordinator. Note that returning latestTableVersion as -1 is acceptable only if the commit
+     * coordinator never ratified any version, i.e. it never accepted any unbackfilled commit.
+     *
+     * @param tablePath The path to the delta log of the table for which the unbackfilled
+     *                  commits should be retrieved.
+     * @param tableConf The table configuration that was returned by the commit coordinator
+     *                  during registration.
+     * @param startVersion The minimum version of the commit that should be returned. Can be null.
+     * @param endVersion The maximum version of the commit that should be returned. Can be null.
+     * @return GetCommitsResponse which has a list of {@link Commit}s and the latestTableVersion
+     *         which is tracked by {@link CommitCoordinatorClientHandler}.
+     */
+    GetCommitsResponse getCommits(
+            String tablePath,
```

Comment:
@prakharjain09 @dhruvarya-db why are we using `logPath` in some places and `tablePath` in otherplaces? Can we take just the `tablePath` everywhere in this interface?

Reply:
+1. This is a miss. We need to fix this everywhere.

Reply:
Should this be fixed immediately in the Kernel-API? The defaults implementation can internally convert one to the other until this is fixed in other places.

Reply:
Will fix it in Kernel-API (everywhere logPath).